### ISSUE TITLE
Identifying the element wrongly when a new dynamic element is added

### ISSUE THREAD

**Issue by sguptatw** [getgauge/taiko#120]
**Expected behavior**
Should be able to identify the element correctly

**Actual behavior**
Identifying the element wrongly when a new element appears dynamically
> Element is identified correctly before the new element appears dynamically
![screenshot-1535585527232](https://user-images.githubusercontent.com/6310197/44821305-8248d480-ac12-11e8-8c86-189c54b45fdd.png)

> Element is identification changes when the new element appears dynamically
![screenshot-1535585714362](https://user-images.githubusercontent.com/6310197/44821324-9a205880-ac12-11e8-8a33-a49a3bf49903.png)

**Steps to replicate**
* Run the script in observe mode
```
(async () => {
    try {
        await openBrowser({headless:false});
        await goto("thetrainline.com");
 
        await write("London",into(inputField(toRightOf("From"))));
        await press("Enter");
        await write("Manchester");
        await press("Enter");
        await click("Tomorrow")
        await click(button({"type":"submit"}));
        await highlight("Continue");
        await click("Continue");
    } catch (e) {
        console.error(e);
    } finally {
        closeBrowser();
    }
 })();
```
> Two elements are getting highlighted for clicking `Continue`. Where as only one should be identified.

**Version**
```
commit - ec0cb85571aa2130abd5fe7a183de1ad709f1b5b
```
> Identification of elements should be done after the ajax calls are completed.

Blocks #28

**Comment by NivedhaSenthil** [comment]
This was also due to flaky wait..fix should be available in fd86932

**Comment by sguptatw** [comment]
The test script is still flaky. Follow the same steps in the original issue. 
When the script is run, taiko is failing to enter the station after loading the page.

Commit - a5da76d97ef5a1e6ca17534424747b96003c8b13

**Comment by Apoorva-GA** [comment]
This should be fixed as part of 888638d275cfebad0747695ebbd1404c60295846. 
Please verify.
